tests/benchmark/utils/s2n_negotiate_client_benchmark.cc

```diff
@@ -0,0 +1,231 @@
+        exit(1);
+    }
+
+    GUARD_EXIT(s2n_connection_set_config(conn, config), "Error setting configuration");
+
+    GUARD_EXIT(s2n_set_server_name(conn, server_name), "Error setting server name");
+
+    GUARD_EXIT(s2n_connection_set_fd(conn, sockfd), "Error setting file descriptor");
+
+    GUARD_EXIT(s2n_connection_set_client_auth_type(conn, S2N_CERT_AUTH_OPTIONAL),
+               "Error setting ClientAuth optional");
+
+    if (use_corked_io) {
+        GUARD_EXIT(s2n_connection_use_corked_io(conn), "Error setting corked io");
+    }
+
+    GUARD_EXIT(s2n_setup_external_psk_list(conn, psk_optarg_list, psk_list_len), "Error setting external psk list");
+
+    if (benchmark_negotiate(conn, sockfd, state, warmup) != 0) {
+        state.SkipWithError("Negotiate Failed\n");
+        if (DEBUG_PRINT) {
+            printf("Error in negotiate!\n");
+        }
+    }
+
+    if (DEBUG_PRINT) {
+        printf("Connected to %s:%s\n", host, port);
+    }
+
+    GUARD_EXIT(s2n_connection_free_handshake(conn), "Error freeing handshake memory after negotiation");
+
+    s2n_blocked_status blocked;
+    int shutdown_rc = s2n_shutdown(conn, &blocked);
+    if (shutdown_rc == -1 && blocked != S2N_BLOCKED_ON_READ) {
+        fprintf(stderr, "Unexpected error during shutdown: '%s'\n", s2n_strerror(s2n_errno, "NULL"));
+        exit(1);
+    }
+
+    GUARD_EXIT(s2n_connection_free(conn), "Error freeing connection");
+
+    GUARD_EXIT(s2n_config_free(config), "Error freeing configuration");
+
+    free(unsafe_verify_data);
+}
+
+static void ClientBenchmark(benchmark::State& state) {
+    int i;
+    for(i = 0; i < WARMUP_ITERS; i++) {
+        client_benchmark(state, true);
+    }
+    int counter = 0;
+    for (auto _ : state) {
+        counter++;
+        if(DEBUG_PRINT){
+            printf("Count: %d\n", counter);
+        }
+        state.PauseTiming();
+        client_benchmark(state, false);
+    }
+}
+
+int Client::start_benchmark_client(int argc, char** argv) {
+    s2n_init();
+    argument_parse(argc, argv);
+
+    char **newv = (char**)malloc((argc + 4) * sizeof(*newv));
+    memmove(newv, argv, sizeof(*newv) * argc);
+    char bench_out[100] = "--benchmark_out=client_";
+    strcat(bench_out, file_prefix);
+    newv[argc] = bench_out;
+    char aggregate[100] = "--benchmark_display_aggregates_only=true";
+    newv[argc+1] = aggregate;
+    newv[argc+2] = bench_format;
+    newv[argc+3] = 0;
+    argc+=3;
+    argv = newv;
+
+    setup_config();
+    unsigned int i;
+    for(i = 0; i < num_suites; i++) {
+        char bench_name[80];
+        strcpy(bench_name, "Client: ");
+        strcat(bench_name, all_suites[i]->name);
+
+        benchmark::RegisterBenchmark(bench_name, ClientBenchmark)->Repetitions(ITERATIONS)->Iterations(1)->Arg(i);
+
+    }
+
+    ::benchmark::Initialize(&argc, argv);
+
+    ::benchmark::RunSpecifiedBenchmarks();
+    free(newv);
+    s2n_cleanup();
+    close(sockfd);
+    freeaddrinfo(ai_list);
```

Comment:
Can this be cleaned up in `setup_config`? I don't think anything needs to use this once the socket is open.